core/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java

```diff
@@ -61,6 +61,7 @@
     private IndexAnalyzers indexAnalyzers;
     private AnalysisRegistry registry;
     private Environment environment;
+    private int maxTokenCount;
```

Comment:
Can you add the following tests to this class es well: 
* test that exceeding the default (10000) throws error, one that uses a request with explain == true and one that doesn't (since this seems to distinguish whether simpleAnalyze or detailedAnalyze gets used internally)
* test with some other low value that overwrites the default
* maybe a test that checks that the default works for the case where "_analyze" is used without specifying an index